### ISSUE TITLE
isosurfaces in glucifer

### ISSUE THREAD

**Issue by RebeccaFarrington** [underworldcode/underworld2#159]
Bec - It would be good to have isosurface.

Owen's response.  Discuss. 

The old isosurface code needs updating to use the functions interface.

You can also now plot dynamic isosurfaces (ie: you can change the isovalue when visualising, not fixed to what you picked when you did the run) with the new volume renderer (glucifer.objects.Volume) but that doesn't yet allow plotting a field with a colour map on the surface, just a solid colour.

The trade-off is the database will be larger when saving a 3D data sample for the volume renderer than just saving a fixed isovalue surface as we used to be able to do.

I guess we need to decide whether we need to either/both:
- Completely pythonise/functionise the old isosurface(and contour?) code
- Adapt new volume isosurface to plot colourmapped fields on surfaces

I have to work on another project solidly for a couple of weeks so can't do much until the end of the month.

**Comment by dansand** [comment]
A possible value-add here: It would be nice to have access the points defining the isosurface. If you have to generate these for the visualisation(?), then perhaps a function that simply hands these coordinates to the user, so they can be used for uw.function evaluations.

**Comment by jmansour** [comment]
Yep, this would be great.  The matplotlib contour tool allows you to access the numpy array of contour path points, but this won't work in parallel.

**Comment by OKaluza** [comment]
I haven't forgotten this, work is under way but to do it properly requires a few underlying design tweaks though, goals:

 * Use FieldSampler to send a sampled grid to LavaVu
 * Add functionality to LavaVu to re-use blocks of data rather than having them locked into the object that created them https://github.com/OKaluza/LavaVu/issues/27.
 * Via this feature, add functionality in LavaVu to take this sampled cube (which currently is only renderable as a volume) and transform it into other types of geometry, eg: slices / isosurfaces
 * Create a python interface to these functions which will also allow accessing the internal data storage of LavaVu geometry as numpy arrays (so: allowing access to the vertices of isosurfaces etc)
 * Create additional objects in glucifer (eg: IsoSurface/Slice) which leverage this interface to transform FieldSampler data into their desired type of geometry and write it back to the database
